ddl/partition.go

```diff
@@ -2373,20 +2398,25 @@ func (w *worker) onReorganizePartition(d *ddlCtx, t *meta.Meta, job *model.Job)
 			// For available state, the new added partition should wait its replica to
 			// be finished, otherwise the query to this partition will be blocked.
 			count := tblInfo.TiFlashReplica.Count
-			needRetry, err := checkPartitionReplica(count, addingDefinitions, d)
+			needRetry, err := checkPartitionReplica(count, count, addingDefinitions, d)
 			if err != nil {
 				// need to rollback, since we tried to register the new
 				// partitions before!
 				return convertAddTablePartitionJob2RollbackJob(d, t, job, err, tblInfo)
 			}
+			// Try for 10 rounds (in case of transient TiFlash issues)
```

Comment:
Why try for 10 rounds instead of `tidb_ddl_error_count_limit `?

Reply:
It is just a number taken from thin air. My idea is that we should not block forever on TiFlash issues, but if it takes too long time, we will skip the wait and add an entry in the error log, so that the DDL can continue at least with the TiKV data. I would assume that if TiFlash cannot replicate the new empty regions, then it will also have problems with replicating current tables regions.
@hehechen what would you suggest? Block until TiFlash is OK, or only wait for 1 copy or as I propose here (maybe with a different number)?

Reply:
> It is just a number taken from thin air. My idea is that we should not block too long on TiFlash issues, but if it takes some time, we will skip the wait and add an entry in the error log, so that the DDL can continue at least with the TiKV data. I would assume that if TiFlash cannot replicate the new empty regions, then it will also have problems with replicating current tables regions. @hehechen what would you suggest? I can remove this, so it will fail after `tidb_ddl_error_count_limit` Or should it wait for `tidb_ddl_error_count_limit` errors and then proceed (including resetting the count)?

If job.ErrorCount > 10, will return `convertAddTablePartitionJob2RollbackJob` and then `onDropTablePartition`(Line 2243) ?

Reply:
> If job.ErrorCount > 10, will return convertAddTablePartitionJob2RollbackJob and then onDropTablePartition(Line 2243) ?

@hehechen Yes.

Did we encounter a case that 'we got some problems when making TiFlash replica, which blocks DDL and it retries a lot of time'?

Reply:
There is an alternative PR, without this change of logic here: https://github.com/pingcap/tidb/pull/42082 Maybe we should merge that one instead, to get all the tests in and wait with extending the logic until we actually see any issues?